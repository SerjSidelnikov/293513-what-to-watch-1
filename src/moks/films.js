export default [
  {
    'id': 0,
    'name': `Fantastic Beasts`,
    'genre': `Comedies`,
    'poster_image': `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    'src': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'preview_video_link': ``,
  },
  {
    'id': 1,
    'name': `Bohemian Rhapsody`,
    'genre': `Documentary`,
    'poster_image': `img/bohemian-rhapsody.jpg`,
    'src': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'preview_video_link': ``,
  },
  {
    'id': 2,
    'name': `Macbeth`,
    'genre': `Dramas`,
    'poster_image': `img/macbeth.jpg`,
    'src': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'preview_video_link': ``,
  },
  {
    'id': 3,
    'name': `Aviator`,
    'genre': `Romance`,
    'poster_image': `img/aviator.jpg`,
    'src': `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    'preview_video_link': ``,
  },
  {
    'id': 4,
    'name': `We need to talk about Kevin`,
    'genre': `Thrillers`,
    'poster_image': `img/we-need-to-talk-about-kevin.jpg`,
    'src': `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    'preview_video_link': ``,
  },
];
