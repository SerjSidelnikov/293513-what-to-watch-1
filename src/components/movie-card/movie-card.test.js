import renderer from 'react-test-renderer';

import MovieCard from './movie-card';
import films from '../../moks/films';

describe(`The application is displayed correctly.`, () => {
  it(`MovieCard screen correctly renders after launch`, () => {
    const handleClick = jest.fn();
    const handleActiveMovie = jest.fn();
    const {title, src} = films[0];

    const component = renderer.create(
        <MovieCard
          title={title}
          src={src}
          onClick={handleClick}
          onActiveMovie={handleActiveMovie}
        />
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
