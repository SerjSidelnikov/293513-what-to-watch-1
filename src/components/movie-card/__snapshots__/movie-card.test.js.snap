// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The application is displayed correctly. MovieCard screen correctly renders after launch 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
      onClick={[MockFunction]}
    >
      
    </a>
  </h3>
</article>
`;
